Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/graph/ActivityNode.java

```diff
@@ -11,30 +12,52 @@
 
 import nl.vu.cs.s2group.nappa.PrefetchingLib;
 import nl.vu.cs.s2group.nappa.prefetchurl.ParameteredUrl;
+import nl.vu.cs.s2group.nappa.room.activity.visittime.AggregateVisitTimeByActivity;
 import nl.vu.cs.s2group.nappa.room.dao.SessionDao;
 import nl.vu.cs.s2group.nappa.room.dao.UrlCandidateDao;
 import nl.vu.cs.s2group.nappa.room.data.ActivityExtraData;
 import nl.vu.cs.s2group.nappa.room.data.SessionData;
 
 public class ActivityNode {
-    private final static String LOG_TAG = ActivityNode.class.getSimpleName();
+    private static final String LOG_TAG = ActivityNode.class.getSimpleName();
 
     public String activityName;
 
+    // TODO Verify possibility of simplifying successor/ancestors structure
+    //  Both `successors` and `ancestors` are defined as a map of `ActivityNode --> visited count`
+    //  for the current session. However, the value of the map doesn't seems to be used, as the
+    //  LiveData object `listSessionAggregateLiveData` provides the aggregate from the past sessions
+    //  and is used in the strategies. In this case, we can likely simply this map to a list
     public Map<ActivityNode, Integer> successors = new ConcurrentHashMap<>();
     public Map<ActivityNode, Integer> ancestors = new ConcurrentHashMap<>();
+    // TODO Refactor LiveData management for SessionAggregate
+    //  This class declares 2 LiveData attributes for the SessionAggregate data:
+    //  * `listSessionAggregateLiveData`: All sessions
+    //  * `listLastNSessionAggregateLiveData`: Last N sessions
+    //  In PR S2-group/NAPPA#67 I modified the PrefetchingLib to instantiate only one of them.
+    //  This class should be refactored to keep a single declaration of the LiveData attribute
+    //  and instantiate the LiveData object in the PrefetchingLib either with the all sessions
+    //  query or with the query of the last N sessions.
+    //  Use the `aggregateVisitTimeLiveData` object as reference.
+    //  This change will require updating the following classes:
+    //  * `NappaUtil#getSuccessorsAggregateVisitFrequency`
+    //  * `PPMPrefetchingStrategy#zeroContextNodes`
+    //  * `PPMWithHITSScoresPrefetchingStrategy#zeroContextNodes`
     private LiveData<List<SessionDao.SessionAggregate>> listSessionAggregateLiveData;
     private LiveData<List<SessionDao.SessionAggregate>> listLastNSessionAggregateLiveData;
     public Map<String, ParameteredUrl> parameteredUrlMap = new HashMap<>();
     public List<ParameteredUrl> parameteredUrlList = new LinkedList<>();            // A list of all parametered URLs within the activity
-    public LiveData<List<UrlCandidateDao.UrlCandidateToUrlParameter>>  urlCandidateDbLiveData;
+    public LiveData<List<UrlCandidateDao.UrlCandidateToUrlParameter>> urlCandidateDbLiveData;
     private LiveData<List<ActivityExtraData>> listActivityExtraLiveData;
-    public float pageRank,authority,hub,authorityS,hubS,prob;
+    public float pageRank, authority, hub, authorityS, hubS, prob;
```

Comment:
Looking at the changes I have formatted this file too.
I should have probably formated the entire project at the beginning, so at least all formation would go in the same commit/PR and stop adding noise afterwards.